### DISCUSSION TITLE
VSC vs Sublime

### DISCUSSION THREAD

**Discussion by romulosardinha** [frontendbr/forum#818] [Perguntas e Respostas]
## Fala ai Fronts!
Quantos ai estão no __Sublime__? E quantos estão no __VSC__? Quantos estão com outro software?

Eu atualmente estou usando mais o VSC pelas ferramentas internas que dá. 
Mas sempre estou namorando a velocidade do Sublime.

**Comment by kivervinicius** [comment]
VSCode for the win!

**Comment by giioohbernini** [comment]
Vim <3

**Comment by TiagoFuelber** [comment]
Uso mais o VSCode, mas as vezes deixo sublime aberto pra codar html, enquanto CSS e PHP no VS.

**Comment by expalmer** [comment]
Sublime e VIM

**Comment by kvnol** [comment]
O que é Sublime?

**Comment by kvnol** [comment]
Brincadeiras a parte, utilizei o Sublime de 2011 até agosto desse ano, trocando pelo VSCode e nunca mais consegui voltar.

**Comment by phsantiago** [comment]
Comecei no Sublime, fui para o VSC, depois VSC com plugin do VIM e hoje não consigo sair do VIM haha

**Comment by matheusml** [comment]
Em 2015 eu comecei a usar duas ferramentas que mudaram a minha vida: React e VSCode.

Hoje, dois anos depois sou apaixonado por ambas. Mas se fosse pra apostar qual delas eu largaria primeiro, certamente seria o React.

VSCode ❤️

**Comment by lBrunol** [comment]
Eu comecei usando o Sublime. Usei por um tempo e gostei bastante. Conheci o VS Code um tempo depois, migrei e não me arrependi. As coisas que me fizeram migrar foram:
- Visual bem enxuto e agradável (sem precisar instalar temas para ficar mais bonito)
- Pequenas features que agilizam e muito o desenvolvimento (isto também tem no sublime)
- Suporte nativo para Typescript e consequentemente, vai muito bem com Angular 2/4
- Debug nativo. O sublime também tem através de extensões, mas quando usei, me pareceu uma grande gambiarra
- Atualizações constantes com melhorias e novas features. Este acho que é o principal motivo pelo qual migrei e ainda utilizo. Sempre tem novos recursos e melhorias, diferente do sublime que, durante o tempo que usei, não recebia muitas atualizações.

**Comment by romulosardinha** [comment]
@matheusml 
O que me levou ao VSCode foi o angular 2, sinto o mesmo, hj uso até mais o react, mas o vscode ganhou lugar no coração mesmo kkk

**Comment by deppbrazil** [comment]
Migrei do:
### Sublime -> VSCode ### 
E adorei... com os plugins então fica uma maravilha 
## <3 ##

**Comment by expalmer** [comment]
Caras, sabe que tentei usar o VS mas meus olhos não deixam, acho que é muito tempo com o Sublime...qual tema vocês usam no VSCODE ?

**Comment by romulosardinha** [comment]
@expalmer 
Não estou usando tema nenhum, ele sozinho já é muito bonito pra mim...

**Comment by expalmer** [comment]
Pois é @rmlSardinha, o meu problema que não gosto de temas muito escuros, força minha vista. Acho o VS bem superior, mas quando fico usando muito me dá um desconforto e volto pro sublime, de repente de seu encontrar um tema mais bacana...vamos ver kk

**Comment by felipefialho** [comment]
Eu virei fã boy declarado do VSCode, editor bom da porra!

**Comment by kivervinicius** [comment]
@expalmer minha solução foi usar o mesmo tema que do sublime

**Comment by Matheus-de-Souza** [comment]
Tirando a confusão com o ícone, o VSCode é o melhor editor, hehe 😄 
Dá pra fazer tudo nele e é levinho comparado às IDEs.
Eu consigo codar nele tudo o que preciso JS, PHP, HTML, CSS/SCSS, C# (Unity) e até fazer o artigo da faculdade em LaTex.

Ah, e o debugger dele é muito bom. Roda até o XDebug do PHP com pouco esforço..

**Comment by expalmer** [comment]
Eu realmente concordo, mas eu gosto do Sublime, porque ele não tem nada, só de um lado as pastas/arquivos e meu código kkkkk Por isso acho que um dia vou pro VIM, eu acho fantástico o VSCODE, mas não gosto de muita informação visual

**Comment by romulosardinha** [comment]
@expalmer então, tem tema igual do sublime, tem até plugin para importar todos os atalhos de teclado do sublime para vc não perder produtividade.

**Comment by lysandroc** [comment]
**o vscode oferece uma gama de plugins irado,** e o melhor é que por ser um projeto open source pode nos garantir que isto vai seguir mais uns anos para frente.

**Comment by gtkatakura-bysoft** [comment]
Eu uso o VSC para trabalhos pessoais e o Sublime Text no trabalho. E existe um único motivo para eu usar o Sublime Text: o repositório tem MUITO CÓDIGO, e é muito comum eu efetuar pesquisas por termos no projeto, e o VSC não dá conta (ele apanha BRUTALMENTE para o Sublime Text em termos de performance em pesquisa). Algumas vezes eu uso o VSC e efetuo pesquisas pelo git grep, mas ainda prefiro o Sublime Text para essas tarefas.

**Comment by giovannism20** [comment]
@gtkatakura-bysoft Já pensou em usar algum editor terminal based ?

**Comment by gtkatakura-bysoft** [comment]
@GiovanniSM20 eu tentei somente com o VIM, mas o VSC e o Sublime Text já são tão produtivos pra mim que não tive paciência de aprender direito o VIM... você tem alguma recomendação?

**Comment by barbier** [comment]
Eu gosto muito do VSCode, mas admito que usaria o Sublime se tivesse uma licença dele

**Comment by vitocaio** [comment]
para PCs mais lentos tem VSCode Online ja :D

**Comment by barraponto** [comment]
Não sei pq o fetiche todo com o Sublime.
É nagware (fica incomodando pedindo pra comprar a licença), o sistema de pacotes dele é confuso, e os suportes de linguagem nativos (que vem instalados) são meio limitados...

Uso sempre o VIM. Tenho evitado VS Code por ser da Microsoft, então pros meus alunos que preferem ferramentas gráficas, indico o Atom. O sistema de pacotes é nativo, então eu só preciso indicar os plugins que eles precisam instalar. Markdown e Github tem suporte nativo. O [Teletype](https://teletype.atom.io/) é ótimo pra pair-programming.

**Comment by leandroaps** [comment]
O Atom é bem pesado, gostava muito do Sublime, e de uns 3 meses pra cá,
tenho usado o VS Code no Linux numa boa, leve, rápido e ótimo suporte,
mesmo sendo MS, ele vale a pena!


-- 

[]´s

*Leandro Ap. de Siqueira*
leandroaps@gmail.com
+55 19 991112239

2018-01-28 9:06 GMT-02:00 Capi Etheriel <notifications@github.com>:

**Comment by mahenrique94** [comment]
VSC +1
